praw/models/reddit/base.py

```diff
@@ -1,25 +1,35 @@
 """Provide the RedditBase class."""
-
-from sys import version_info
+from six import string_types
 
 from ...const import API_PATH, urlparse
 from ...exceptions import ClientException, PRAWException
 from ..base import PRAWBase
 
-if version_info.major > 2:
-    basestring = str  # pylint: disable=invalid-name
-
 
 class RedditBase(PRAWBase):
     """Base class that represents actual Reddit objects."""
 
+    REPR_FIELD = 'id'
+    STR_FIELD = 'fullname'
+
     @staticmethod
     def _url_parts(url):
         parsed = urlparse(url)
         if not parsed.netloc:
             raise ClientException('Invalid URL: {}'.format(url))
         return parsed.path.rstrip('/').split('/')
 
+    # pylint: disable=invalid-name
+    @property
+    def id(self):
+        """Return the object's id."""
+        try:
```

Comment:
Shouldn't `id` work as it with the backwards compatible mode available? When we break backwards compatibility, I think having people access id via `object.a.id` is fine. I also think at such a time that`fullname` should be removed. I think I added it for simplicity with things that depend on it, but you're right it really doesn't belong.

Reply:
`object.id` will work either way, but `object.fullname` of course won’t work if `fullname` isn’t defined here because `fullname` isn’t a real reddit attribute.

Defining `id` here isn’t redundant though: defining it here would allow it show up in the tab completion (for `object.<TAB>`). I think that `id` should be in the tab completion for `object` if doing `object.id` is to be recommended over `object.a.id`.

> I think having people access id via `object.a.id` is fine.

It’s fine, but I think it’s uglier than `object.id`. I doubt many others would like the look of `object.a.id` either.

> I think I added it for simplicity with things that depend on it, but you're right it really doesn't belong.

I didn’t say `fullname` shouldn’t belong here (it kind of has to because it’s not a real attribute) but having `object.fullname` and then recommending `object.a.id` instead of `object.id` feels like an unfortunate inconsistency. They’re similar in value and it makes sense for them to go together.